src/main/java/eu/arrowhead/blacklist/BlacklistSystemInfo.java

```diff
@@ -34,127 +42,215 @@ public String getSystemName() {
 				.build();
+	}
 
-		// monitor
+	//-------------------------------------------------------------------------------------------------
+	private ServiceModel getMonitorServiceModel() {
+		final List<InterfaceModel> monitorInterfaces = new ArrayList<>();
 
-		final HttpOperationModel echo = new HttpOperationModel.Builder()
+		final HttpOperationModel httpEchoOp = new HttpOperationModel.Builder()
 				.method(HttpMethod.GET.name())
 				.path(BlacklistConstants.HTTP_API_OP_ECHO)
 				.build();
 
-		final HttpInterfaceModel monitorInterface = new HttpInterfaceModel.Builder(templateName, getDomainAddress(), getServerPort())
+		final HttpInterfaceModel httpMonitorIntf = new HttpInterfaceModel.Builder(httpTemplateName, getDomainAddress(), getServerPort())
 				.basePath(BlacklistConstants.HTTP_API_MONITOR_PATH)
-				.operation(Constants.SERVICE_OP_ECHO, echo)
+				.operation(Constants.SERVICE_OP_ECHO, httpEchoOp)
 				.build();
 
-		final ServiceModel monitor = new ServiceModel.Builder()
+		monitorInterfaces.add(httpMonitorIntf);
+
+		if (isMqttApiEnabled()) {
+			final MqttInterfaceModel mqttMonitorIntf = new MqttInterfaceModel.Builder(mqttTemplateName, getMqttBrokerAddress(), getMqttBrokerPort())
+					.baseTopic(BlacklistConstants.MQTT_API_MONITOR_BASE_TOPIC)
+					.operations(Set.of(Constants.SERVICE_OP_ECHO))
+					.build();
+			monitorInterfaces.add(mqttMonitorIntf);
+
+		}
+
+		return new ServiceModel.Builder()
 				.serviceDefinition(Constants.SERVICE_DEF_MONITOR)
 				.version(BlacklistConstants.VERSION_MONITOR)
 				.metadata(BlacklistConstants.METADATA_KEY_UNRESTRICTED_DISCOVERY, false)
-				.serviceInterface(monitorInterface)
+				.serviceInterfaces(monitorInterfaces)
 				.build();
-
-
-		return List.of(discovery, management, generalManagement, monitor);
 	}
-
-	//-------------------------------------------------------------------------------------------------
-	@Override
-	public SystemModel getSystemModel() {
-		if (systemModel == null) {
-			SystemModel.Builder builder = new SystemModel.Builder()
-					.address(getAddress())
-					.version(Constants.AH_FRAMEWORK_VERSION);
-
-			if (AuthenticationPolicy.CERTIFICATE == this.getAuthenticationPolicy()) {
-				builder = builder.metadata(Constants.METADATA_KEY_X509_PUBLIC_KEY, getPublicKey());
-			}
-
-			systemModel = builder.build();
-		}
-
-		return systemModel;
+	
+	protected PublicConfigurationKeysAndDefaults getPublicConfigurationKeysAndDefaults() {
+		return new PublicConfigurationKeysAndDefaults(
+				Set.of(Constants.SERVER_ADDRESS,
+						Constants.SERVER_PORT,
+						Constants.MQTT_API_ENABLED,
+						Constants.DOMAIN_NAME,
+						Constants.AUTHENTICATION_POLICY,
+						Constants.ENABLE_MANAGEMENT_FILTER,
+						Constants.MANAGEMENT_POLICY,
+						Constants.ENABLE_BLACKLIST_FILTER,
+						Constants.FORCE_BLACKLIST_FILTER,
+						Constants.ALLOW_SELF_ADDRESSING,
+						Constants.ALLOW_NON_ROUTABLE_ADDRESSING,
+						Constants.MAX_PAGE_SIZE,
+						Constants.SERVICE_ADDRESS_ALIAS,
```

Comment:
I don't think Blacklist has this property, but Constants.SERVICEREGISTRY_ADDRESS and Constants.SERVICEREGISTRY_PORT has. 

Also not sure in self addressing, non routable addressing. 
Here just collect those that you actually use in blacklist system (and they are considered public).